doc/user/content/sql/system-catalog.md

```diff
@@ -249,6 +249,19 @@ Field          | Type       | Meaning
 `schema_id`    | [`bigint`] | The ID of the schema to which the sink belongs.
 `name`         | [`text`]   | The name of the sink.
 
+### `mz_source_info`
+
+The `mz_source_info` table contains a row for each source in the system.
+
+Field          | Type       | Meaning
+---------------|------------|----------
+`source_name`  | [`text`]   | Materialize's internal name for the source.
+`source_id`    | [`text`]   | Materialize's unique ID for the source. Equivalent to `mz_sources.id`.
+`dataflow_id`  | [`bigint`] | The ID of the dataflow responsible for processing this source / partition.
+`partition_id` | [`text`]   | The ID of the partition assigned to this source / dataflow.
```

Comment:
This sounds backwards to me: the source has many partitions, and there are various, like, `SourcePartitionWorker`s that get assigned partition IDs. But we're not mentioning the concept of `SourcePartitionWorker`s here, so I think we should just say something like "The ID of the partition within the source." Maybe a few sentences about what a partition ID is, if you like? "This corresponds to a topic partition when using Kafka sources. Source types which do not support partitions, like file sources, will report (idk what do they report) as the partition ID."

Reply:
I'm happy to keep working on this one, as partition ID is a weird one. I ended up saying: "Partition assignments are defined in a source specific manner."

Looking at the implementations, this is how they are displayed:

```rust
impl fmt::Display for PartitionId {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        match self {
            PartitionId::Kafka(id) => write!(f, "{}", id.to_string()),
            PartitionId::S3 => write!(f, "s3"),
            PartitionId::Kinesis => write!(f, "kinesis"),
            PartitionId::File => write!(f, "0"),
        }
    }
}
```

So files show `0`, Kafka shows `0..N` and the other sources show the name of the source. Simple sources won't show up at all.